l10n_ch_account_reconcile_isr/models/bank_statement.py

```diff
@@ -0,0 +1,95 @@
+                            )
+                            AND company_id = %(company)s
+                """
+        params = {
+            'company': self.env.user.company_id.id,
+        }
+        if statements:
+            sql_query += ' AND stl.statement_id IN %(statement)s'
+            params['statement'] = (tuple(statements.ids),)
+        sql_query += ' ORDER BY stl.id'
+        self.env.cr.execute(sql_query, params)
+        st_lines_left = self.env[
+            'account.bank.statement.line'
+        ].browse([line.get('id') for line in self.env.cr.dictfetchall()])
+
+        # try to assign partner to bank_statement_line
+        stl_to_assign_partner = [
+            stl.id for stl in st_lines_left if not stl.partner_id
+        ]
+        refs = list(set([
+            st.name for st in st_lines_left if not st.partner_id
+        ]))
+        acc_cr = st_lines_left[0].journal_id.default_credit_account_id
+        acc_dt = st_lines_left[0].journal_id.default_debit_account_id
+        if st_lines_left and stl_to_assign_partner and refs\
+           and acc_cr and acc_dt:
+
+            sql_query = self._prepare_query()
+            params.update({
+                'accounts': (acc_cr.id, acc_dt.id),
+                'refs': tuple(refs)
+            })
+
+            if statements:
+                sql_query += 'AND stl.id IN %(statement)s'
+                params['statement'] = tuple(stl_to_assign_partner)
+            self.env.cr.execute(sql_query, params)
+            results = self.env.cr.dictfetchall()
+            st_line = self.env['account.bank.statement.line']
+            for line in results:
+                st_line.browse(
+                    line.get('id')
+                ).write({'partner_id': line.get('partner_id')})
+
+        return {
+            'st_lines_ids': st_lines_left.ids,
+            'notifications': [],
+            'statement_name':
+                len(statements) == 1 and statements[0].name or False,
+            'num_already_reconciled_lines': 0,
+        }
+
+    def _prepare_query(self):
+        """Replace ref by transaction_ref to match aml"""
+        sql_query = """SELECT aml.partner_id, aml.transaction_ref, stl.id
+                FROM account_move_line aml
+                    JOIN account_account acc ON acc.id = aml.account_id
+                    JOIN account_bank_statement_line stl
+                    ON aml.transaction_ref = stl.name
+                WHERE (aml.company_id = %(company)s
+                    AND aml.partner_id IS NOT NULL)
+                    AND (
+                        (aml.statement_id IS NULL
+                            AND aml.account_id IN %(accounts)s)
+                        OR
+                        (acc.internal_type IN ('payable', 'receivable')
+                            AND aml.reconciled = false)
+                        )
+                    AND aml.transaction_ref IN %(refs)s
+                    OR  aml.ref IN %(refs)s
```

Comment:
Is this condition needed when you want to match on `transaction_ref` ?

Reply:
I updated it with call to super, so now it's not broken
As for different ways to reconcile, there is no option to choose, to do so is better to split this  addon on base and other base will add settings, other will register ways to reconcile for now if you install it will apply everywhere